### PR TITLE
bf: S3C-1427 use expired interval to avg out throughput

### PR DESCRIPTION
**Pull request by philipyoo**
When an interval of data in Redis expires, throughput will
abruptly reduce. Given the data we collect, we can only
calculate the average throughput. To ease the erratic
decrease on expiration of an interval, instead, get the
average of elapsed time for the newest interval and
remaining time of the interval multiplied against the
average of the just-expired interval.

In Redis, we need to save an extra interval to reference
the just-expired data.

### CODE REVIEW

**Comment by rahulreddy** [comment]
Are there any tests associated with this?

**Comment by philipyoo** [comment]
@ironman-machine
RUN_BACKBEAT_CRR_TESTS=1
try

**Comment by philipyoo** [comment]
@ironman-machine
RUN_BACKBEAT_CRR_TESTS=1
try

**Comment by philipyoo** [comment]
@ironman-machine try

**Comment by philipyoo** [comment]
@ironman-machine
DEFAULT_BRANCH=z/1.0
RUN_BACKBEAT_CRR_TESTS=true
try

**Comment by philipyoo** [comment]
@ironman-machine
DEFAULT_BRANCH=master
RUN_BACKBEAT_CRR_TESTS=true
try

**Comment by philipyoo** [comment]
@ironman-machine
RUN_BACKBEAT_CRR_TESTS=true
try

**Comment by philipyoo** [comment]
@ironman-machine 
RUN_BACKBEAT_CRR_TESTS=true
SCALITY_INTEGRATION_BRANCH=ultron/master
REPO_NAME=Arsenal
SCALITY_ARSENAL_BRANCH=bf/S3C-1427-avgThroughputBetweenExpiredIntervals
try

**Comment by philipyoo** [comment]
@ironman-machine
RUN_BACKBEAT_CRR_TESTS=true
SCALITY_INTEGRATION_BRANCH=ultron/master
REPO_NAME=Arsenal
SCALITY_ARSENAL_BRANCH=bf/S3C-1427-avgThroughputBetweenExpiredIntervals
DEFAULT_BRANCH=z/1.0
try

**Comment by philipyoo** [comment]
@ironman-machine
RUN_BACKBEAT_CRR_TESTS=true
SCALITY_INTEGRATION_BRANCH=ultron/master
REPO_NAME=Arsenal
SCALITY_ARSENAL_BRANCH=bf/S3C-1427-avgThroughputBetweenExpiredIntervals
DEFAULT_BRANCH=z/1.0
try

File: tests/unit/metrics/StatsModel.js [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+            },
+        ], done);
+    });
+
+    it('should correctly record a 500 on the server', done => {
+        statsModel.report500(id, (err, res) => {
+            assert.ifError(err);
+
+            const expected = [[null, 1], [null, 1]];
+            assert.deepEqual(res, expected);
```
Comment (jonathan-gramain) [inline tests/unit/metrics/StatsModel.js]: `deepStrictEqual`?

Comment (philipyoo) [reply tests/unit/metrics/StatsModel.js]: oops

**Pull request merged**